### ISSUE TITLE
Debian 10 - Lando performance randomly degrades while in use

### ISSUE THREAD

**Issue by cchoe1** [lando/lando#1935]
Lando version: v3.0.0-rc23

My landofile (details removed)

```
name: mysite
recipe: pantheon
config:
  framework: drupal8
  env: dev
  site: mysite
  id: ID
  xdebug: true
tooling:
  drush:
    cmd: /app/vendor/bin/drush
proxy:
  pma:
    - pma.my-site.lndo.site
services:
  appserver:
    type: php
    xdebug: true
    config:
      xdebug: true
  database:
    portforward: 32111
    healthcheck:
  pma:
    type: phpmyadmin
    hosts:
      - database
```

Computer specs:
 - Ryzen 3700x 8-core (16 hyperthreaded) CPU
 - 32GB DDR4 3000mhz RAM
 - /dev/sda1 = 120GB SSD
 - /dev/sdc1 = 240GB SSD

Currently on Debian Buster.  I have my lando project in a secondary internal SSD drive (`sdc`) which is mounted onto my root (/) partition which is contained on drive `sda`.  Something will trigger `lando drush cr` to jump from 3.5 seconds to 45-70 seconds.  My computer is extremely beefy so 3.5 seconds sounds like the normal speed while 45-70 seconds seems to be the result of *some* bug.

I've also tried setting up my project directly onto drive `sda` thus completing cutting off `sdc` from the picture.  It doesn't do anything so I don't think this has anything to do with the drives I'm using.  All drives use ext4 for the filesystem.

I can reset this anomaly by unmounting /dev/sdc1 (my secondary drive) and then remounting it onto a *different* directory (still within my root partition).  Once I do this and restart lando (no rebuild necessary) a `lando drush cr` will take 3-4 seconds again and my site will run lightning fast.  If I remount it onto the same directory as before, there will be no performance change.  The site continues to run lightning fast despite multiple cache-rebuilds, database imports.  Then, all of a sudden and without warning, the cache rebuilds will go from 3s to 70s.

I've done a bit of surface-level debugging using `docker stats`.  In the lightning fast state with `lando drush cr`, the `appserver` container will utilize about 50-60% of my CPU while the `cache` container will use about 40-50%.  In the slow state, the `appserver` container will sit at 99% CPU utilization without the `cache` ever stepping in.

I'm very much unsure of how lando works under the hood but I'd presume this has something to do with the caching container.  Something is causing it to get 'stuck' or unusable or otherwise out-of-commission at some point and then my cache rebuilds will drop to incredibly slow speeds.  

 - Edited the title to be more precise

**Comment by cchoe1** [comment]
Also I should note, I ran this exact same project on my Ubuntu machine a few months prior before getting my current computer.  The ubuntu machine ran very quickly and reliably without any issues like this ever cropping up.  The same configuration is being used on my Debian machine and I periodically run into this issue.

**Comment by cchoe1** [comment]
I think I found the issue.  It appears to be xdebug slowing down cache clears.

I read this issue here: https://github.com/lando/lando/issues/1668

Running `time lando drush cr` with xdebug enabled outputs:

real    0m46.141s
user    0m0.407s
sys     0m0.069s

Adding the pre-drush and post-drush events for killing xdebug before a drush command, re-enabling xdebug, then running `time lando drush cr` outputs the following:

real    0m5.951s
user    0m2.637s
sys     0m0.321s

It appears there was a bug when restarting the app in a different directory, xdebug would not get started even though my landofile has it enabled by default.  This is why I was seeing huge improvements in speed.  I would then figure out my breakpoints weren't working and I'd rebuild the project and it'd then get turned on, but my subsequent requests would then be very slow.

I did verify that remounting + `lando start` would not turn on xdebug by:
 1. Remount drive in new directory
 2. `lando start`
 3. `lando ssh` into appserver
 4. `ls /usr/local/etc/php/conf.d/docker-php-ext-xdebug.ini` // does not exist

Once I turned on xdebug, performance slowed down.  I'm still unsure why my other computer would not run across this issue though because both machines would have xdebug on permanently in dev environments.  However, my newer machine is seeing much larger discrepancies when turning on/off xdebug.

This isn't a new issue so I'll close but perhaps this can also raise attention to the performance hit that xdebug can bring about.
